src/config/verification.rs

```diff
@@ -159,7 +164,56 @@ fn build_verification_options_from_flags(
     Ok(Some(verification_config))
 }
 
-pub(crate) async fn build_sigstore_trust_root() -> Result<Option<Arc<SigstoreTrustRoot>>> {
+/// Function that builds the Sigstore trust root used for verification. If a trust-config flag is
+/// provided, it uses that file to load the trust root. Otherwise, it builds the trust root from
+/// Sigstore's TUF repository.
+pub(crate) async fn build_sigstore_trust_root(
+    matches: &ArgMatches,
```

Comment:
I would change the signature of the function to receive an `Option<PathBuf>`. In this way it's clear what has to be provided when invoking the function, plus testing would become easier.